### ISSUE TITLE
Async PV for workspace: enabling feature (toggle)

### ISSUE THREAD

**Issue by benoitf** [eclipse-che/che#15960]
### Enable feature
subtask of https://github.com/eclipse/che/issues/15384 

By default workspaces are always using a PVC (or `emptyDir` for ephemeral workspaces)

In the same approach used for ephemeral we could add a new attribute of the workspace definition

```yaml
attributes:
  asyncPersist: true
```

attribute name can be changed.
 
By using that toggle, it means that a new container will be added to the workspace/pod which will do the sync. And of course, it will use emptyDir as well

It will allow to have this feature enabled per workspace to try how it works and be able to compare, etc.

**Comment by ibuziuk** [comment]
Proposed UD design -  an experimental checkbox that could be enabled **only when** temporary storage toggle is on:

![image](https://user-images.githubusercontent.com/1461122/83248819-ef055000-a1a5-11ea-9373-44dbbbb7be3a.png)

**Comment by ibuziuk** [comment]
cc @beaumorley any comments ^

**Comment by benoitf** [comment]
if you check async but you disable temporary storage IMHO it doesn't work as well so this is why I was thinking of a dropdown

**Comment by ibuziuk** [comment]
@benoitf my idea was that checkbox can only be enabled (maybe even visible) only if the temporary toggle is on

**Comment by ibuziuk** [comment]
After the discussion during prio call it looks like it would be better to opt for dropdown: 
![image](https://user-images.githubusercontent.com/1461122/83278852-82557a00-a1d4-11ea-8f8a-127ed2bd1429.png)

**Comment by vparfonov** [comment]
Demo: https://youtu.be/EU_4M1ZHELw

**Comment by ibuziuk** [comment]
@vparfonov do we have a separate issue for UD customizations?
if yes, I believe we can close this one once https://github.com/eclipse/che/pull/16972 is merged
